### PR TITLE
Condvars

### PR DESCRIPTION
**Pull request by piscisaureus**
Created a branch joyent/libuv@condvars for PR #567, so I could add some fixups.

The Windows side of things is good to merge now, @bnoordhuis can you give a final thumbs up on the unix part?

### CODE REVIEW

**Comment by bnoordhuis** [comment]
LGTM

File: include/uv.h [hunk 1]
```diff
@@ -1800,6 +1800,36 @@ UV_EXTERN int uv_fs_event_init(uv_loop_t* loop, uv_fs_event_t* handle,
...
+ * Note:
+ * 1. callers should be prepared to deal with spurious wakeups.
+ * 2. the granularity of timeout on Windows is never less than one millisecond.
+ * 3. uv_cond_timedwait takes a relative timeout, not an absolute time.
+ * 4. the precision of timeout on OSX is never less than one microsecond.
+ *    Here is the reason.
+ *    OSX doesn't support CLOCK_MONOTONIC nor pthread_condattr_setclock()
+ *    (see man pthread_cond_init on OSX).
+ *    An example in man pthread_cond_timedwait on OSX uses gettimeofday()
+ *    and its resolution is a microsecond.
```
Comment (bnoordhuis) [inline include/uv.h:1828]: This should be more explicit about the clock skew issue and that it only pertains to OS X. But it's not a show-stopper, we'll address it in a follow-up commit.

**Comment by piscisaureus** [comment]
Cool. @hnakamur Can you sign the CLA? http://nodejs.org/cla.html

**Comment by hnakamur** [comment]
@piscisaureus Yes. I've just signed the CLA.

**Comment by piscisaureus** [comment]
Thanks a lot! Landed in 976c8a4

**Pull request closed without merging**